### DISCUSSION TITLE
Modbus communication Orno 515

### DISCUSSION THREAD

**Discussion by chrissooo** [aldas/modbus-tcp-client#123] [Q&A]
Hi,
for first thank you for the great tool. I use it at the moment to communicate with my Photovoltaik WR over TCP to get some values. I used one of the examples to get it to Work. It's work great. I store the values in InfluxDB for Grafana visuallosatiom.

Now I try to communicate with a another Device, but over Modbus RTU with a USB to Modebus Stick. (/dev/...)

[](https://b2b.orno.pl/download-resource/26063/) 

For example I will get the current now with address A000H.

Can some help me how I can get it? No clue how I can make it.

**Comment by chrissooo** [comment]
Here are the link to the documentation: [Orno 515 documentation ](https://b2b.orno.pl/download-resource/26063/)

**Reply by aldas** [reply]
I assume that address `A000` is in hex so `40960` as dec. Manual says that `Current total active power` is `ULong` so 8 bytes.  8 bytes is 4 registers.

With higher level API you can create packets like that.
```php
$fc3requests = ReadRegistersBuilder::newReadHoldingRegisters($address, $unitID)
    ->uint64(0xA000, 'current_total_active_power')
    ->build(); 
```
In  that way we create packet to fetch these 8 bytes (4 registers) and after that we can use higher level API to extract correct data type into assoc array. 

I have released `3.1.0` with some RTU related improvements. There is new example for RTU over TCP + higher level API [examples/rtu_over_tcp_with_higherlevel_api.php](https://github.com/aldas/modbus-tcp-client/blob/master/examples/rtu_over_tcp_with_higherlevel_api.php). If you are dealing with serial devices you can combine it with `examples/rtu_usb_to_serial.php` example.
